{"expand":"renderedFields,names,schema,transitions,operations,editmeta,changelog","id":"12382903","self":"https://issues.apache.org/jira/rest/api/latest/issue/12382903","key":"DERBY-3217","fields":{"issuetype":{"self":"https://issues.apache.org/jira/rest/api/2/issuetype/1","id":"1","description":"A problem which impairs or prevents the functions of the product.","iconUrl":"https://issues.apache.org/jira/images/icons/issuetypes/bug.png","name":"Bug","subtask":false},"timespent":null,"project":{"self":"https://issues.apache.org/jira/rest/api/2/project/10594","id":"10594","key":"DERBY","name":"Derby","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/projectavatar?pid=10594&avatarId=10122","24x24":"https://issues.apache.org/jira/secure/projectavatar?size=small&pid=10594&avatarId=10122","16x16":"https://issues.apache.org/jira/secure/projectavatar?size=xsmall&pid=10594&avatarId=10122","32x32":"https://issues.apache.org/jira/secure/projectavatar?size=medium&pid=10594&avatarId=10122"},"projectCategory":{"self":"https://issues.apache.org/jira/rest/api/2/projectCategory/10090","id":"10090","description":"DB related projects","name":"DB"}},"fixVersions":[{"self":"https://issues.apache.org/jira/rest/api/2/version/12313111","id":"12313111","description":"","name":"10.4.1.3","archived":false,"released":true,"releaseDate":"2008-04-24"}],"aggregatetimespent":null,"resolution":{"self":"https://issues.apache.org/jira/rest/api/2/resolution/1","id":"1","description":"A fix for this issue is checked into the tree and tested.","name":"Fixed"},"customfield_12310220":"2007-11-21 09:31:58.155","customfield_12312322":null,"customfield_12312323":null,"customfield_12310420":"23499","customfield_12310222":"1_*:*_1_*:*_4903920199_*|*_5_*:*_1_*:*_0","customfield_12312320":null,"customfield_12312321":null,"customfield_12312120":null,"customfield_12312121":null,"resolutiondate":"2008-01-16T12:21:30.385+0000","workratio":-1,"customfield_12312328":null,"customfield_12312329":null,"customfield_12312326":null,"customfield_12310300":null,"customfield_12312327":null,"customfield_12312324":null,"customfield_12312325":null,"lastViewed":null,"watches":{"self":"https://issues.apache.org/jira/rest/api/2/issue/DERBY-3217/watchers","watchCount":0,"isWatching":false},"created":"2007-11-20T18:09:30.186+0000","priority":{"self":"https://issues.apache.org/jira/rest/api/2/priority/3","iconUrl":"https://issues.apache.org/jira/images/icons/priorities/major.png","name":"Major","id":"3"},"labels":[],"customfield_12312333":null,"customfield_12312334":null,"customfield_12310310":"2.0","customfield_12312331":null,"customfield_12312332":null,"timeestimate":null,"aggregatetimeoriginalestimate":null,"versions":[{"self":"https://issues.apache.org/jira/rest/api/2/version/12313111","id":"12313111","description":"","name":"10.4.1.3","archived":false,"released":true,"releaseDate":"2008-04-24"}],"customfield_12311120":null,"customfield_12312330":null,"issuelinks":[],"customfield_12312339":null,"assignee":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=vemund","name":"vemund","emailAddress":"vemund at sun dot com","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Vemund Ã˜stgaard","active":true},"customfield_12312337":null,"customfield_12312338":null,"updated":"2011-01-21T17:51:04.479+0000","customfield_12312335":null,"customfield_12311720":null,"customfield_12312336":null,"status":{"self":"https://issues.apache.org/jira/rest/api/2/status/6","description":"The issue is considered finished, the resolution is correct. Issues which are closed can be reopened.","iconUrl":"https://issues.apache.org/jira/images/icons/statuses/closed.png","name":"Closed","id":"6","statusCategory":{"self":"https://issues.apache.org/jira/rest/api/2/statuscategory/3","id":3,"key":"done","colorName":"green","name":"Complete"}},"components":[{"self":"https://issues.apache.org/jira/rest/api/2/component/11413","id":"11413","name":"Test"}],"timeoriginalestimate":null,"description":"Getting the following junit tests failures on just on the j9 jvm  platform running on windows:\nThere were 11 failures:\n1) testDefault(org.apache.derbyTesting.functionTests.tests.engine.ErrorStreamTest)junit.framework.AssertionFailedError: shutdown engine not implemened\n\tat org.apache.derbyTesting.junit.DataSourceConnector.shutEngine(DataSourceConnector.java:130)\n\tat org.apache.derbyTesting.junit.TestConfiguration.shutdownEngine(TestConfiguration.java:1091)\n\tat org.apache.derbyTesting.functionTests.tests.engine.ErrorStreamTest.testDefault(ErrorStreamTest.java:128)\n\tat java.lang.reflect.AccessibleObject.invokeV(AccessibleObject.java:205)\n\tat <unknown class>.<unknown method>(Unknown Source)\n\tat org.apache.derbyTesting.junit.BaseTestCase.runBare(BaseTestCase.java:95)\n2) testFile(org.apache.derbyTesting.functionTests.tests.engine.ErrorStreamTest)junit.framework.AssertionFailedError: shutdown engine not implemened\n\tat org.apache.derbyTesting.junit.DataSourceConnector.shutEngine(DataSourceConnector.java:130)\n\tat org.apache.derbyTesting.junit.TestConfiguration.shutdownEngine(TestConfiguration.java:1091)\n\tat org.apache.derbyTesting.functionTests.tests.engine.ErrorStreamTest.testFile(ErrorStreamTest.java:141)\n\tat java.lang.reflect.AccessibleObject.invokeV(AccessibleObject.java:205)\n\tat <unknown class>.<unknown method>(Unknown Source)\n\tat org.apache.derbyTesting.junit.BaseTestCase.runBare(BaseTestCase.java:95)\n3) testWrongFile(org.apache.derbyTesting.functionTests.tests.engine.ErrorStreamTest)junit.framework.AssertionFailedError: shutdown engine not implemened\n\tat org.apache.derbyTesting.junit.DataSourceConnector.shutEngine(DataSourceConnector.java:130)\n\tat org.apache.derbyTesting.junit.TestConfiguration.shutdownEngine(TestConfiguration.java:1091)\n\tat org.apache.derbyTesting.functionTests.tests.engine.ErrorStreamTest.testWrongFile(ErrorStreamTest.java:160)\n\tat java.lang.reflect.AccessibleObject.invokeV(AccessibleObject.java:205)\n\tat <unknown class>.<unknown method>(Unknown Source)\n\tat org.apache.derbyTesting.junit.BaseTestCase.runBare(BaseTestCase.java:95)\n4) testMethod(org.apache.derbyTesting.functionTests.tests.engine.ErrorStreamTest)junit.framework.AssertionFailedError: shutdown engine not implemened\n\tat org.apache.derbyTesting.junit.DataSourceConnector.shutEngine(DataSourceConnector.java:130)\n\tat org.apache.derbyTesting.junit.TestConfiguration.shutdownEngine(TestConfiguration.java:1091)\n\tat org.apache.derbyTesting.functionTests.tests.engine.ErrorStreamTest.testMethod(ErrorStreamTest.java:179)\n\tat java.lang.reflect.AccessibleObject.invokeV(AccessibleObject.java:205)\n\tat <unknown class>.<unknown method>(Unknown Source)\n\tat org.apache.derbyTesting.junit.BaseTestCase.runBare(BaseTestCase.java:95)\n5) testWrongMethod(org.apache.derbyTesting.functionTests.tests.engine.ErrorStreamTest)junit.framework.AssertionFailedError: shutdown engine not implemened\n\tat org.apache.derbyTesting.junit.DataSourceConnector.shutEngine(DataSourceConnector.java:130)\n\tat org.apache.derbyTesting.junit.TestConfiguration.shutdownEngine(TestConfiguration.java:1091)\n\tat org.apache.derbyTesting.functionTests.tests.engine.ErrorStreamTest.testWrongMethod(ErrorStreamTest.java:198)\n\tat java.lang.reflect.AccessibleObject.invokeV(AccessibleObject.java:205)\n\tat <unknown class>.<unknown method>(Unknown Source)\n\tat org.apache.derbyTesting.junit.BaseTestCase.runBare(BaseTestCase.java:95)\n6) testField(org.apache.derbyTesting.functionTests.tests.engine.ErrorStreamTest)junit.framework.AssertionFailedError: shutdown engine not implemened\n\tat org.apache.derbyTesting.junit.DataSourceConnector.shutEngine(DataSourceConnector.java:130)\n\tat org.apache.derbyTesting.junit.TestConfiguration.shutdownEngine(TestConfiguration.java:1091)\n\tat org.apache.derbyTesting.functionTests.tests.engine.ErrorStreamTest.testField(ErrorStreamTest.java:217)\n\tat java.lang.reflect.AccessibleObject.invokeV(AccessibleObject.java:205)\n\tat <unknown class>.<unknown method>(Unknown Source)\n\tat org.apache.derbyTesting.junit.BaseTestCase.runBare(BaseTestCase.java:95)\n7) testWrongField(org.apache.derbyTesting.functionTests.tests.engine.ErrorStreamTest)junit.framework.AssertionFailedError: shutdown engine not implemened\n\tat org.apache.derbyTesting.junit.DataSourceConnector.shutEngine(DataSourceConnector.java:130)\n\tat org.apache.derbyTesting.junit.TestConfiguration.shutdownEngine(TestConfiguration.java:1091)\n\tat org.apache.derbyTesting.functionTests.tests.engine.ErrorStreamTest.testWrongField(ErrorStreamTest.java:236)\n\tat java.lang.reflect.AccessibleObject.invokeV(AccessibleObject.java:205)\n\tat <unknown class>.<unknown method>(Unknown Source)\n\tat org.apache.derbyTesting.junit.BaseTestCase.runBare(BaseTestCase.java:95)\n8) testFileOverMethod(org.apache.derbyTesting.functionTests.tests.engine.ErrorStreamTest)junit.framework.AssertionFailedError: shutdown engine not implemened\n\tat org.apache.derbyTesting.junit.DataSourceConnector.shutEngine(DataSourceConnector.java:130)\n\tat org.apache.derbyTesting.junit.TestConfiguration.shutdownEngine(TestConfiguration.java:1091)\n\tat org.apache.derbyTesting.functionTests.tests.engine.ErrorStreamTest.testFileOverMethod(ErrorStreamTest.java:257)\n\tat java.lang.reflect.AccessibleObject.invokeV(AccessibleObject.java:205)\n\tat <unknown class>.<unknown method>(Unknown Source)\n\tat org.apache.derbyTesting.junit.BaseTestCase.runBare(BaseTestCase.java:95)\n9) testFileOverField(org.apache.derbyTesting.functionTests.tests.engine.ErrorStreamTest)junit.framework.AssertionFailedError: shutdown engine not implemened\n\tat org.apache.derbyTesting.junit.DataSourceConnector.shutEngine(DataSourceConnector.java:130)\n\tat org.apache.derbyTesting.junit.TestConfiguration.shutdownEngine(TestConfiguration.java:1091)\n\tat org.apache.derbyTesting.functionTests.tests.engine.ErrorStreamTest.testFileOverField(ErrorStreamTest.java:278)\n\tat java.lang.reflect.AccessibleObject.invokeV(AccessibleObject.java:205)\n\tat <unknown class>.<unknown method>(Unknown Source)\n\tat org.apache.derbyTesting.junit.BaseTestCase.runBare(BaseTestCase.java:95)\n10) testFileOverMethodAndField(org.apache.derbyTesting.functionTests.tests.engine.ErrorStreamTest)junit.framework.AssertionFailedError: shutdown engine not implemened\n\tat org.apache.derbyTesting.junit.DataSourceConnector.shutEngine(DataSourceConnector.java:130)\n\tat org.apache.derbyTesting.junit.TestConfiguration.shutdownEngine(TestConfiguration.java:1091)\n\tat org.apache.derbyTesting.functionTests.tests.engine.ErrorStreamTest.testFileOverMethodAndField(ErrorStreamTest.java:302)\n\tat java.lang.reflect.AccessibleObject.invokeV(AccessibleObject.java:205)\n\tat <unknown class>.<unknown method>(Unknown Source)\n\tat org.apache.derbyTesting.junit.BaseTestCase.runBare(BaseTestCase.java:95)\n11) testMethodOverField(org.apache.derbyTesting.functionTests.tests.engine.ErrorStreamTest)junit.framework.AssertionFailedError: shutdown engine not implemened\n\tat org.apache.derbyTesting.junit.DataSourceConnector.shutEngine(DataSourceConnector.java:130)\n\tat org.apache.derbyTesting.junit.TestConfiguration.shutdownEngine(TestConfiguration.java:1091)\n\tat org.apache.derbyTesting.functionTests.tests.engine.ErrorStreamTest.testMethodOverField(ErrorStreamTest.java:326)\n\tat java.lang.reflect.AccessibleObject.invokeV(AccessibleObject.java:205)\n\tat <unknown class>.<unknown method>(Unknown Source)\n\tat org.apache.derbyTesting.junit.BaseTestCase.runBare(BaseTestCase.java:95)\n\n\n\n","customfield_10010":null,"timetracking":{},"customfield_12312026":null,"customfield_12310200":[{"self":"https://issues.apache.org/jira/rest/api/2/customFieldOption/10369","value":"Regression Test Failure","id":"10369"}],"customfield_12312023":null,"customfield_12312024":null,"customfield_12312340":null,"aggregatetimeestimate":null,"customfield_12312022":null,"customfield_12312341":null,"customfield_12312220":null,"customfield_12310921":null,"customfield_12310920":"37368","summary":"ErrorStreamTest failing on ibm j9 jvm","creator":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=mikem","name":"mikem","emailAddress":"mikem_app at sbcglobal dot net","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Mike Matrigali","active":true},"subtasks":[],"customfield_12310090":null,"reporter":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=mikem","name":"mikem","emailAddress":"mikem_app at sbcglobal dot net","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Mike Matrigali","active":true},"customfield_12310291":null,"customfield_12310290":null,"aggregateprogress":{"progress":0,"total":0},"environment":"J9 - VM for the Java(TM) platform, Version 2.3\n(c) Copyright IBM Corp. 1991, 2006  All Rights Reserved\nTarget: 20061023_08962_lHdSMR (Windows XP 5.1 build 2600 Service Pack 2 x86)\nJIT  - 20060629_1804ifx1_r8","customfield_12311020":null,"customfield_12310050":null,"duedate":null,"customfield_12310250":null,"progress":{"progress":0,"total":0},"comment":{"startAt":0,"maxResults":13,"total":13,"comments":[{"self":"https://issues.apache.org/jira/rest/api/2/issue/12382903/comment/12543978","id":"12543978","author":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=mikem","name":"mikem","emailAddress":"mikem_app at sbcglobal dot net","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Mike Matrigali","active":true},"body":"I am seeing junit failures in this test when run on j9.  Unfortunately the machine running these\r\ntests had a number of problems so I don't know exactly when the issue started, but it has been happening at least \r\nsince 11/12/2007.","updateAuthor":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=mikem","name":"mikem","emailAddress":"mikem_app at sbcglobal dot net","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Mike Matrigali","active":true},"created":"2007-11-20T18:13:17.324+0000","updated":"2007-11-20T18:13:17.324+0000"},{"self":"https://issues.apache.org/jira/rest/api/2/issue/12382903/comment/12543986","id":"12543986","author":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=mikem","name":"mikem","emailAddress":"mikem_app at sbcglobal dot net","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Mike Matrigali","active":true},"body":"IBM jvm results are available from:\r\nhttp://db.apache.org/derby/derby_tests.html\r\nhttp://people.apache.org/~fuzzylogic/derby_test_results/\r\n\r\nwindows results are pretty consistently posted, work is continuing on getting the linux results posted.\r\n\r\nSee following for first known example:\r\nhttp://people.apache.org/~fuzzylogic/derby_test_results/main/derbymain/testlog/weme6.1/594394-suites.All_diff.txt\r\n\r\nSee following which shows failure continuous from 11/12\r\nhttp://people.apache.org/~fuzzylogic/derby_test_results/main/derbymain/weme6.1_history.html","updateAuthor":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=mikem","name":"mikem","emailAddress":"mikem_app at sbcglobal dot net","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Mike Matrigali","active":true},"created":"2007-11-20T18:22:00.613+0000","updated":"2007-11-20T18:22:00.613+0000"},{"self":"https://issues.apache.org/jira/rest/api/2/issue/12382903/comment/12544390","id":"12544390","author":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=vemund","name":"vemund","emailAddress":"vemund at sun dot com","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Vemund Ã˜stgaard","active":true},"body":"The test shuts down the engine to be able to reset the properties that control the derby error stream. When running in this jsr169 configuration connections are done through DataSource, and the DataSourceConnector in the test framework does not support shutting down the engine. I don't know why you can't shut down the engine through DataSourceConnector, but I am sure it's for a good reason.\r\n\r\nSo, I don't know how the test could be rewritten to not shut down the engine, or to avoid DataSource in this configuration, so I guess the only way forward is to avoid running this test with configurations that do not support DriverManager. Unless anyone has other suggestions I will go ahead and implement that change, requiring JDBC2 or 3 for this test to run.","updateAuthor":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=vemund","name":"vemund","emailAddress":"vemund at sun dot com","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Vemund Ã˜stgaard","active":true},"created":"2007-11-21T09:31:58.155+0000","updated":"2007-11-21T09:31:58.155+0000"},{"self":"https://issues.apache.org/jira/rest/api/2/issue/12382903/comment/12544519","id":"12544519","author":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=bryanpendleton","name":"bryanpendleton","emailAddress":"bpendleton dot derby at gmail dot com","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Bryan Pendleton","active":true},"body":"I'm confused about why the test framework doesn't support\r\nshutting down the engine. Is this a limitation of DataSourceConnector?\r\nOr of DataSource itself? ","updateAuthor":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=bryanpendleton","name":"bryanpendleton","emailAddress":"bpendleton dot derby at gmail dot com","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Bryan Pendleton","active":true},"created":"2007-11-21T16:22:10.392+0000","updated":"2007-11-21T16:22:10.392+0000"},{"self":"https://issues.apache.org/jira/rest/api/2/issue/12382903/comment/12544745","id":"12544745","author":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=vemund","name":"vemund","emailAddress":"vemund at sun dot com","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Vemund Ã˜stgaard","active":true},"body":"I don't know. All I see is that DataSourceConnector does this: \r\n\r\n    public void shutEngine() throws SQLException {\r\n        Assert.fail(\"shutdown engine not implemened\");\r\n    }\r\n\r\n\r\nWhile DriverManagerConnector does: \r\n\r\n    public void shutEngine() throws SQLException {\r\n        \r\n        getConnectionByAttributes(\"jdbc:derby:\", \"shutdown\", \"true\");        \r\n    }\r\n\r\nI figured it was because DataSource did not support shutting down the engine.\r\n","updateAuthor":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=vemund","name":"vemund","emailAddress":"vemund at sun dot com","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Vemund Ã˜stgaard","active":true},"created":"2007-11-22T09:42:39.406+0000","updated":"2007-11-22T09:44:56.164+0000"},{"self":"https://issues.apache.org/jira/rest/api/2/issue/12382903/comment/12544779","id":"12544779","author":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=johnemb","name":"johnemb","emailAddress":"johnemb at gmail dot com","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"John H. Embretsen","active":true},"body":"As far as I can see, there is no documented way to shut down the Derby engine when you are using a DataSource (I've only looked at EmbeddedSimpleDataSource (used for Java ME) and EmbeddedDataSource). \r\n\r\nHowever, I found that it is actually possible, by first setting the databaseName to the empty String, then setting the connection attribute \"shutdown=true\", and finally trying to obtain a connection. The exception is the same as I get when shutting down the engine using DriverManager.\r\n\r\n        // Create an EmbeddedSimpleDataSource, setDatabaseName, setCreateDatabase, getConnection...\r\n\r\n        try {\r\n\t    derbyDs.setDatabaseName(\"\");\r\n\t    derbyDs.setConnectionAttributes(\"shutdown=true\");\r\n\t    System.out.println(\"Shutting down engine...?\");\r\n            derbyDs.getConnection();\r\n        } catch (SQLException ex) {\r\n\t\tdo {\r\n\t\t    System.err.println(\"----- SQLException -----\");\r\n\t\t    System.err.println(\"SQLState:   \" + ex.getSQLState());\r\n\t\t    System.err.println(\"Error code: \" + ex.getErrorCode());\r\n\t\t    //ex.printStackTrace();\r\n\t\t    ex = ex.getNextException();\r\n\t\t} while (ex != null);\r\n            }\r\n        }\r\n\r\n\r\nTo restart the engine, create a new datasource object and set the correct databaseName, then getConnection().\r\nIt seems that the engine is started on the first call to getConnection().\r\n\r\nThe test harness is a bit more complicated (less direct) in the way it sets DataSource properties, but it should be possible to implement something like this there as well, if needed. ","updateAuthor":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=johnemb","name":"johnemb","emailAddress":"johnemb at gmail dot com","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"John H. Embretsen","active":true},"created":"2007-11-22T11:10:40.746+0000","updated":"2007-11-22T11:10:40.746+0000"},{"self":"https://issues.apache.org/jira/rest/api/2/issue/12382903/comment/12545483","id":"12545483","author":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=vemund","name":"vemund","emailAddress":"vemund at sun dot com","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Vemund Ã˜stgaard","active":true},"body":"Thank you for the tip, John.\r\n\r\nBased on your suggestion I've implemented the shutEngine() method in DataSourceConnector, and I can now run the test successfully in a Java ME configuration.\r\n\r\nI'll do some more testing in other configurations and submit a patch for this tomorrow.","updateAuthor":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=vemund","name":"vemund","emailAddress":"vemund at sun dot com","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Vemund Ã˜stgaard","active":true},"created":"2007-11-26T15:47:26.898+0000","updated":"2007-11-26T15:47:26.898+0000"},{"self":"https://issues.apache.org/jira/rest/api/2/issue/12382903/comment/12545754","id":"12545754","author":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=vemund","name":"vemund","emailAddress":"vemund at sun dot com","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Vemund Ã˜stgaard","active":true},"body":"This patch implements the shutEngine() method for DataSourceConnector, enabling the ErrorStream test to run successfully also on Java ME platforms.\r\n\r\nI haven't tested the patch on j9, but it works on the phoneME advanced platform.","updateAuthor":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=vemund","name":"vemund","emailAddress":"vemund at sun dot com","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Vemund Ã˜stgaard","active":true},"created":"2007-11-27T08:41:34.602+0000","updated":"2007-11-27T08:41:34.602+0000"},{"self":"https://issues.apache.org/jira/rest/api/2/issue/12382903/comment/12548367","id":"12548367","author":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=dyret","name":"dyret","emailAddress":"Dyre dot Tjeldvoll at oracle dot com","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Dyre Tjeldvoll","active":true},"body":"Is anyone planning to take this patch for a spin on J9? If not, I plan to commit it tomorrow (Wednesday 5 Dec). (I trust that a followup-patch can be made if problems surface on J9 later).","updateAuthor":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=dyret","name":"dyret","emailAddress":"Dyre dot Tjeldvoll at oracle dot com","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Dyre Tjeldvoll","active":true},"created":"2007-12-04T19:21:22.877+0000","updated":"2007-12-04T19:21:22.877+0000"},{"self":"https://issues.apache.org/jira/rest/api/2/issue/12382903/comment/12548518","id":"12548518","author":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=myrna","name":"myrna","emailAddress":"m dot v dot lunteren at gmail dot com","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Myrna van Lunteren","active":true},"body":"I took the patch for a test drive, and at least test engine.ErrorStreamTest passes fine with the patch; but it actually also passes  for me without the patch. Maybe I'd need to run suites.All to see the problem. I think a follow-up patch to address later problems  is ok.","updateAuthor":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=myrna","name":"myrna","emailAddress":"m dot v dot lunteren at gmail dot com","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Myrna van Lunteren","active":true},"created":"2007-12-05T02:33:40.051+0000","updated":"2007-12-05T02:33:40.051+0000"},{"self":"https://issues.apache.org/jira/rest/api/2/issue/12382903/comment/12548694","id":"12548694","author":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=dyret","name":"dyret","emailAddress":"Dyre dot Tjeldvoll at oracle dot com","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Dyre Tjeldvoll","active":true},"body":"Committed revision 601361","updateAuthor":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=dyret","name":"dyret","emailAddress":"Dyre dot Tjeldvoll at oracle dot com","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Dyre Tjeldvoll","active":true},"created":"2007-12-05T15:28:56.685+0000","updated":"2007-12-05T15:28:56.685+0000"},{"self":"https://issues.apache.org/jira/rest/api/2/issue/12382903/comment/12559471","id":"12559471","author":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=dyret","name":"dyret","emailAddress":"Dyre dot Tjeldvoll at oracle dot com","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Dyre Tjeldvoll","active":true},"body":"Fix checked in","updateAuthor":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=dyret","name":"dyret","emailAddress":"Dyre dot Tjeldvoll at oracle dot com","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Dyre Tjeldvoll","active":true},"created":"2008-01-16T12:21:30.369+0000","updated":"2008-01-16T12:21:30.369+0000"},{"self":"https://issues.apache.org/jira/rest/api/2/issue/12382903/comment/12582564","id":"12582564","author":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=dyret","name":"dyret","emailAddress":"Dyre dot Tjeldvoll at oracle dot com","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Dyre Tjeldvoll","active":true},"body":"This issue has no fix version, but has been checked in to trunk after the 10.3 branch was cut, and has not been merged to the 10.3 branch (before the last release). \r\n(Some have been checked into trunk after 10.4 was cut and then merged to 10.4).","updateAuthor":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=dyret","name":"dyret","emailAddress":"Dyre dot Tjeldvoll at oracle dot com","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Dyre Tjeldvoll","active":true},"created":"2008-03-27T09:36:46.951+0000","updated":"2008-03-27T09:36:46.951+0000"}]},"votes":{"self":"https://issues.apache.org/jira/rest/api/2/issue/DERBY-3217/votes","votes":0,"hasVoted":false},"worklog":{"startAt":0,"maxResults":20,"total":0,"worklogs":[]},"customfield_12311820":"0|i06rdb:"}}